server/application_api: TestStatusAPIContentionEvents failed

server/application_api.TestStatusAPIContentionEvents [failed](https://mesolite.cluster.engflow.com/invocations/default/4261988e-e6b0-4c8c-b7d9-8455e1d74c5e?testReportRun=3&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on release-25.4.10-rc @ [71a8b899a3473eec94e8fd2291c69249f02d347a](https://github.com/cockroachdb/cockroach/commits/71a8b899a3473eec94e8fd2291c69249f02d347a):


```
=== RUN   TestStatusAPIContentionEvents
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestStatusAPIContentionEvents2101534234
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    contention_test.go:120: 
        	Error Trace:	pkg/server/application_api/contention_test.go:120
        	Error:      	Should be true
        	Test:       	TestStatusAPIContentionEvents
        	Messages:   	expect to find contention event for table 106, but found {Events:{IndexContentionEvents:[tableID=11 indexID=3
        	            	  num contention events: 1
        	            	  cumulative contention time: 94.458533ms
        	            	  keys:
        	            	    /Table/11/3/"\x80"/12/1/"\x01\x01\x80\xca\xfd8\xb4\xdfEIs\xa1)\x8as\xc4o\x9fc"/0 contending txns:
        	            	      id=161e331b-ae41-43e3-9752-92af88055153 count=1
        	            	] NonSQLKeysContention:[]} Errors:[]}
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestStatusAPIContentionEvents2101534234
--- FAIL: TestStatusAPIContentionEvents (40.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #168226 server/application_api: TestStatusAPIContentionEvents failed [C-test-failure O-robot T-observability branch-release-25.4 release-blocker]
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusAPIContentionEvents.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63231

--- a/pkg/server/application_api/contention_test.go
+++ b/pkg/server/application_api/contention_test.go
@@ -106,28 +106,26 @@ SET TRACING=off;
 	sqlstatstestutil.WaitForTransactionEntriesAtLeast(t, obsConn, 1,
 		sqlstatstestutil.TransactionFilter{App: "contentionTest"})
 
-	var resp serverpb.ListContentionEventsResponse
-	require.NoError(t,
-		srvtestutils.GetStatusJSONProtoWithAdminOption(
+	testutils.SucceedsSoon(t, func() error {
+		var resp serverpb.ListContentionEventsResponse
+		if err := srvtestutils.GetStatusJSONProtoWithAdminOption(
 			s2,
 			"contention_events",
 			&resp,
-			true /* isAdmin */),
-	)
-
-	require.GreaterOrEqualf(t, len(resp.Events.IndexContentionEvents), 1,
-		"expecting at least 1 contention event, but found none")
-
-	found := false
-	for _, event := range resp.Events.IndexContentionEvents {
-		if event.TableID == descpb.ID(testTableID) && event.IndexID == descpb.IndexID(1) {
-			found = true
-			break
+			true, /* isAdmin */
+		); err != nil {
+			return err
 		}
-	}
-
-	require.True(t, found,
-		"expect to find contention event for table %d, but found %+v", testTableID, resp)
+		for _, event := range resp.Events.IndexContentionEvents {
+			if event.TableID == descpb.ID(testTableID) && event.IndexID == descpb.IndexID(1) {
+				return nil
+			}
+		}
+		return errors.Newf(
+			"contention event for table %d not found in response: %+v",
+			testTableID, resp,
+		)
+	})
 
 	server1Conn.CheckQueryResultsRetry(t, `
   SELECT count(*)

